tests/load/test_checkquery.py

```diff
@@ -0,0 +1,359 @@
+
+
+@patch("subprocess.run")
+def test_changed_queries_failure(mock_run, capsys):
+    mock_result = MagicMock()
+    mock_result.configure_mock(**{"returncode": 1, "stderr": "no git"})
+
+    mock_run.return_value = mock_result
+    assert changed_queries() is None
+
+    err_out = capsys.readouterr().err
+    assert "ERROR: no git" == err_out.strip()
+
+
+# all_queries
+@pytest.mark.parametrize(
+    "tree, expected",
+    [
+        (
+            [
+                ("/root", ("src",), ("README.txt",)),
+                ("/root/src", (), ("spam.sh", "eggs.py")),
+            ],
+            [],
+        ),
+        (
+            [
+                ("/root", ("src",), ("a.sparql",)),
+                ("/root/src", (), ("sparql.pdf", "b.sparql")),
+            ],
+            [
+                QueryFile(Path("/root/a.sparql")),
+                QueryFile(Path("/root/src/b.sparql")),
+            ],
+        ),
+    ],
+)
+def test_all_queries(tree, expected):
+    with patch("os.walk") as mock_walk:
+        mock_walk.return_value = tree
+        assert all_queries() == expected
+
+
+# execute
+
+
+def test_execute(a_query):
+    with pytest.raises(QueryExecutionException) as err:
+        _ = execute(a_query, 1, None, 0)
+
+        assert str(err) == f"{a_query.path} : Failed too many times."
+
+
+# check_limit
+@pytest.mark.parametrize(
+    "candidate, limit",
+    [
+        ("1", 1),
+        ("34", 34),
+        ("1000", 1000),
+    ],
+)
+def test_check_limit_pos(candidate, limit):
+    assert check_limit(candidate) == limit
+
+
+@pytest.mark.parametrize(
+    "candidate",
+    [
+        "0",
+        "-1",
+        "a",
+        "word",
+    ],
+)
+def test_check_limit_neg(candidate):
+    with pytest.raises(argparse.ArgumentTypeError) as err:
+        _ = check_limit(candidate)
+
+    assert str(err.value) == "LIMIT must be an integer of value 1 or greater."
+
+
+# check_timeout
+@pytest.mark.parametrize(
+    "candidate, timeout",
+    [
+        ("1", 1),
+        ("9", 9),
+        ("8888", 8888),
+    ],
+)
+def test_check_timeout_pos(candidate, timeout):
+    assert check_timeout(candidate) == timeout
+
+
+@pytest.mark.parametrize(
+    "candidate",
+    [
+        "0",
+        "-1",
+        "O",
+        "ten",
+    ],
+)
+def test_check_timeout_neg(candidate):
+    with pytest.raises(argparse.ArgumentTypeError) as err:
+        _ = check_timeout(candidate)
+
+    assert str(err.value) == "timeout must be an integer of value 1 or greater."
+
+
+# main
+
+
+@pytest.mark.parametrize("arg", ["-h", "--help"])
+def test_main_help(arg):
+    with pytest.raises(SystemExit) as err:
+        _ = main(arg)
+        assert err.code == 0
+
+
+@pytest.mark.parametrize(
+    "args",
+    [
+        ["-a", "-c"],
+        ["-p", "-f"],
+        ["-p", "-c"],
+        ["-p", "-a"],
+        ["-c", "-p", "-a"],
+        ["-c", "-f", "-a"],
+    ],
+)
+def test_main_mutex_opts(args):
+    """some options cannot be used together"""
```

Comment:
Minor nit on the comment: capital S and a period at the end. I think for doctrings it also makes sense to have them always like:

```python
"""
Docstring starting on the next line between the quotes.
"""
```

Reply:
Done. Changes made to all docstrings.